src/components/ha-dialog.ts

```diff
@@ -78,6 +78,7 @@ export class HaDialog extends MwcDialog {
           color: inherit;
         }
         .header_title {
+          color: var(--primary-text-color);
```

Comment:
Why is this needed now and not before?
```suggestion
          color: var(--mdc-dialog-heading-ink-color);
```

Reply:
Your requested change does not work. It would once again result in a black header color since `--mdc-dialog-heading-ink-color` is not defined, which is weird since it is set here: https://github.com/home-assistant/frontend/blob/349a5f52b10691416962ac707deaf5cf3ce030af/src/resources/styles.ts#L291

It for example gets set directly for one other media player related dialog (in `hui-dialog-web-browser-play-media.ts`), but I think it would be better to set it once for all dialogs in `.mdc-dialog`.

No idea why no one came across this so far. I myself only recently switched to the dark default theme.

Reply:
This was working before, as like you linked `mdc-dialog-heading-ink-color` is defined and should be used, so something changed that broke it. We should find that.\

The overruling in `hui-dialog-web-browser-play-media` should be removed.

Reply:
I think the problem could come from the fact this hides the previous `ha-dialog` style:
https://github.com/home-assistant/frontend/blob/e63a78bcdb556dcb1e64605ec4145ca7e956e12a/src/panels/media-browser/hui-dialog-select-media-player.ts#L79

Reply:
Not sure what you mean by that?

Reply:
Is this only an issue for this specific dialog, or for every dialog?

Reply:
That dialog doesn't use the `haStyleDialog` styles, that is the issue. You should add those styles to `static get styles()`

Reply:
Thanks for pointing that out!